packages/contracts/test/invariant/diamond/facets/PoolFacetHandler.sol

```diff
@@ -0,0 +1,266 @@
+     * @param delay The number of blocks to set as the redemption delay.
+     */
+    function setRedemptionDelay(uint256 delay) public {
+        vm.prank(admin);
+        ubiquityPoolFacet.setRedemptionDelayBlocks(delay);
+    }
+
+    /**
+     * @notice Manipulates the minting and redemption fees for UbiquityPoolFacet.
+     * @dev This function allows the admin to set the fees for minting and redeeming tokens in the pool.
+     * It assumes the caller is the admin and pranks the transaction as the admin to set the fees.
+     * The function also ensures that the provided mint and redeem fees fall within the acceptable range.
+     * @param mintFee The fee to be set for minting, expressed in basis points (1/100 of a percent).
+     * @param redeemFee The fee to be set for redeeming, expressed in basis points (1/100 of a percent).
+     */
+    function setMintAndRedeemFees(uint256 mintFee, uint256 redeemFee) public {
+        vm.assume(mintFee >= 100000 && mintFee <= 200000);
+        vm.assume(redeemFee >= 100000 && redeemFee <= 200000);
+
+        vm.prank(admin);
+        ubiquityPoolFacet.setFees(0, mintFee, redeemFee);
+    }
+
+    function collectRedemption() public {
+        ubiquityPoolFacet.collectRedemption(0);
+    }
+
+    /**
+     * @notice Manipulates the pool ceiling for UbiquityPoolFacet.
+     * @dev This function allows the admin to set a new ceiling for the pool, which determines the maximum
+     * amount of collateral that can be utilized in the pool. The function assumes the caller is the admin
+     * and pranks the transaction as the admin to set the new ceiling.
+     * @param newCeiling The new ceiling value to be set for the pool, representing the maximum amount
+     * of collateral in the pool.
+     */
+    function setPoolCeiling(uint256 newCeiling) public {
+        vm.prank(admin);
+        ubiquityPoolFacet.setPoolCeiling(0, newCeiling);
+    }
+
+    /**
+     * @notice Manipulates the collateral price and updates the corresponding collateral ratio.
+     * @dev This function adjusts the price of the collateral token using a mock ChainLink price feed.
+     * It assumes the new price is within the allowed range and updates the collateral ratio in the UbiquityPoolFacet.
+     * @param _newPrice The new price of the collateral, scaled by 1e8 (e.g., a price of $1 is represented as 1e8).
+     */
+    function setCollateralPrice(int256 _newPrice) public {
+        vm.assume(_newPrice >= 50_000_000 && _newPrice <= 200_000_000);
+
+        collateralTokenPriceFeed.updateMockParams(
+            1, // round id
+            _newPrice,
+            block.timestamp, // started at
+            block.timestamp, // updated at
+            1 // answered in round
+        );
+
+        ubiquityPoolFacet.updateChainLinkCollateralPrice(0);
+
+        uint256 newCollateralRatio = uint256(1e6 * 1e8).div(uint256(_newPrice));
+        ubiquityPoolFacet.setCollateralRatio(newCollateralRatio);
```

Comment:
Also may revert in some cases because collateral ratio can't be > `1_000_000`. Why do we even need to call `setCollateralRatio()` on setting chainlink related prices [here](https://github.com/ubiquity/ubiquity-dollar/blob/dfff4f314386ef0ec4d0fb6091367623823480e0/packages/contracts/test/invariant/diamond/facets/PoolFacetHandler.sol#L157) and [here](https://github.com/ubiquity/ubiquity-dollar/blob/dfff4f314386ef0ec4d0fb6091367623823480e0/packages/contracts/test/invariant/diamond/facets/PoolFacetHandler.sol#L178)?